app/controllers/pet_applications_controller.rb

```diff
@@ -0,0 +1,31 @@
+class PetApplicationsController < ApplicationController
+
+  def create
+    applicant = Application.find(params[:id])
+    pet_application = PetApplication.create!({
+    pet_id: params[:pet_id],
+    application_id: params[:id],
+    status: "Pending",
+    })
+  
+    redirect_to "/applications/#{applicant.id}"
+  end
+
+  def update
+    pet_application = PetApplication.find_pet_application(params[:id], params[:pet_id])
+    pet_application.update!(status: params[:status])
+    
+    application = Application.find(params[:id])
+    if application.all_pet_statuses.all?("Approved")
+      application.update!({
+        status: "Approved"
+      })
+    elsif application.all_pet_statuses.any?("Rejected")
+    application.update!({
+      status: "Rejected"
+    })
+    end 
+
+    redirect_to "/admin/applications/#{params[:id]}"
```

Comment:
Per your question, I do believe that you would include the update on an application with any association to a pet in this controller. Then the `update` action in your `applications` controller is left to merely update anything specific to an application with no association to a pet.